code/game/objects/items/storage/pouch.dm

```diff
@@ -479,3 +481,24 @@
 	draw_mode = 0
 	can_hold = list("/obj/item/ammo_magazine/handful",
 					)
+
+/obj/item/storage/pouch/shotgun/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/ammo_magazine/shotgun))
+		var/obj/item/ammo_magazine/shotgun/M = W
+		if(M.current_rounds)
+			if(contents.len < storage_slots)
+				to_chat(user, "<span class='notice'>You start refilling [src] with [M].</span>")
+				if(!do_after(user, 15, TRUE, 5, BUSY_ICON_GENERIC)) 
+					return
+				for(var/x = 1 to storage_slots)
+					handle_item_insertion(M.create_handful(), 1, user)
```

Comment:
```suggestion
					if(!handle_item_insertion(M.create_handful(), 1, user))
						break
```
Or, if you want to avoid proc calls in if checks:
```
var/cont
for(var/x = 1 to storage_slots)
	cont = handle_item_insertion(M.create_handful(), 1, user)
	if(!cont)
		break
```